usr/sbin/p11sak/p11sak.c

```diff
@@ -2477,7 +2519,7 @@ static CK_RV generate_asymmetric_key(CK_SESSION_HANDLE session, CK_SLOT_ID slot,
             fprintf(stderr, "Error parsing EC parameters!\n");
             goto done;
         }
-    } else {
+    } else if (kt != kt_IBM_DILITHIUM) {
```

Comment:
looks like you removed the final else case in none of the key types match. Don't we still need that?

... actually a switch() for all 3 case (+default case) would probably look better.

Reply:
else if (kt != kt_IBM_DILITHIUM) is the original else. For every other key_type the error is thrown, but for kt_IBM_DILITHIUM I dont want the error and I dont need a read_ibm_dilithium_args() function

Reply:
Though I can still move from the if else construction to a switch

Reply:
changed to switch however